db/migrate/20170703153415_dedupe_more_specialist_publisher_change_notes.rb

```diff
@@ -0,0 +1,29 @@
+class DedupeMoreSpecialistPublisherChangeNotes < ActiveRecord::Migration[5.1]
+  def up
+    dupe_change_note_ids = ChangeNote.group(:public_timestamp, :note)
+                                     .having("COUNT(*) > 1")
+                                     .pluck("array_agg(id)")
+
+
+    document_ids = ChangeNote.where(id: dupe_change_note_ids.flatten,
+                                    edition_id: nil)
+                             .pluck(:document_id).flatten.uniq
+
+    # ~ 572 specialist-publisher change notes.
+    ChangeNote.where(id: dupe_change_note_ids.flatten,
+                     edition_id: nil).delete_all
+
+
+    # Remove Edition#details[:change_history] and represent the Edition
+    # downstream, the history will be reconstructed from ChangeNotes.
+    Edition.where(document_id: document_ids).each do |edition|
+      edition_details = edition.details
+      edition_details.delete(:change_history)
+      edition.update!(details: edition_details)
+
+      if Rails.env.production?
+        Commands::V2::RepresentDownstream.new.call(edition.content_id)
```

Comment:
I think we can pass this an array of IDs, it might be quicker doing that than calling it in each iteration.